elementpath/xpath2/_xpath2_functions.py

```diff
@@ -668,9 +667,9 @@ def deep_equal():
                 return False
             elif value1 is None:
                 return True
-            elif (is_xpath_node(value1)) ^ (is_xpath_node(value2)):
+            elif (isinstance(value1, XPathNode)) ^ (isinstance(value2, XPathNode)):
```

Comment:
`!=` seems more readable here than `^`. (Even just the time it takes me to understand whether both do the same thing is a reason to simplify this.)

Reply:
`^` should be needed in this case for let the method works.